ethcore/trace/src/db.rs

```diff
@@ -57,6 +57,34 @@ impl Key<FlatBlockTraces> for H256 {
 	}
 }
 
+/// `DatabaseExtras` provides an interface to query extra data which is not stored in TraceDB,
+/// but necessary to work correctly.
+pub trait DatabaseExtras {
+	/// Returns hash of given block number.
+	fn block_hash(&self, block_number: BlockNumber) -> Option<H256>;
+
+	/// Returns hash of transaction at given position.
+	fn transaction_hash(&self, block_number: BlockNumber, tx_position: usize) -> Option<H256>;
+}
+
+impl<T: BlockProvider> DatabaseExtras for T {
+	fn block_hash(&self, block_number: BlockNumber) -> Option<H256> {
+		self.block_hash( block_number)
```

Comment:
Does it actually call the right version of `block_hash`? Or just recursively calls itself?

Reply:
I just assumed the tests would ensure that, but looking at the test mocks we have I am doubtful too, so best be explicit. Good catch.

Reply:
I think rust compiler gives a warning in such cases, so hopefuly we're fine. But indeed might be worth a test or at least explicit notation.

Reply:
I think it does too, but to clear up the doubt (and save any readers some brain-friction): https://github.com/paritytech/parity-ethereum/pull/10868/commits/2d06a4e89aa5ca0cc014ad972ab4634cfe1ce65c